src/lib/ndpi_main.c

```diff
@@ -5788,17 +5788,6 @@ ndpi_protocol ndpi_detection_giveup(struct ndpi_detection_module_struct *ndpi_st
 	confidence = NDPI_CONFIDENCE_DPI_PARTIAL;
       }
 
-      if(flow->host_server_name[0] != '\0') {
-        ndpi_protocol_match_result ret_match;
-
-        ndpi_match_host_subprotocol(ndpi_str, flow, (char *) flow->host_server_name,
-				    strlen((const char *) flow->host_server_name), &ret_match,
-				    NDPI_PROTOCOL_DNS);
-
-        if(ret_match.protocol_id != NDPI_PROTOCOL_UNKNOWN)
-          guessed_host_protocol_id = ret_match.protocol_id;
-      }
-
```

Comment:
I am glad you removed that block as it does not make sense to me and should be handled in the dissector (if required).

Although this was added with 6edb7bedd7b5c1eb0060f5d503f9e37a6d58c086 the commit message does not help me much with that specific code block.